app/models/ext_management_system.rb

```diff
@@ -931,14 +931,17 @@ def perf_capture_enabled?
   alias_method :perf_capture_enabled, :perf_capture_enabled?
   Vmdb::Deprecation.deprecate_methods(self, :perf_capture_enabled => :perf_capture_enabled?)
 
-  ###################################
-  # Event Monitor
-  ###################################
-
+  # Some workers hold open a connection to the provider and thus do not
+  # automatically pick up authentication changes.  These workers have to be
+  # restarted manually for the new credentials to be used.
   def after_update_authentication
     stop_event_monitor_queue_on_credential_change
```

Comment:
What calls stop_refresh_worker_queue_on_change and stop_refresh_worker_queue_on_credential_change.  For that matter, what calls stop_event_monitor_queue_on_change ?

Reply:
We only need to stop the refresh worker if they hold a blocking connection, that is currently only vmware and k8s/openshift.  Those providers call `stop_refresh_worker_queue_on_credential_change`

`stop_(refresh_worker|event_monitor)_queue_on_change` is called as an `after_save` callback